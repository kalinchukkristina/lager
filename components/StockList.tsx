import React, { useEffect } from "react";
import { View, Text } from "react-native";
import { Base } from "../styles";
import productModel from './../models/products';


export default function StockList({ products, setProducts }) {
    useEffect(async () => {
      setProducts(await productModel.getProducts());
    }, []);
  
    const list = products.map((product, index) => {
      return <Text 
              key={index} 
              style={Base.prod}
              >
                { product.name } - { product.stock }
              </Text>
    });
  
    return (
      <View>
        {list}
      </View>
    );
  }