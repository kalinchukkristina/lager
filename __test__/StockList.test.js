/**
 * I Lagerförteckningen ska det finnas en lista med produkter. Listan ska innehålla produktens namn och lagersaldo.
 * Kolla om StockList komponent innehåller alltid samma data och har inget ovändate beteende;
 */

import { render } from '@testing-library/react-native';
import StockList from '../components/StockList';

const products = [
    { name: "Tulip", stock: 60 },
    { name: "Daisy", stock: 202 },
    { name: "Peony", stock: 35 },
];

const setProducts = () => false;

test('List should contain three items', async () => {
    const { getByText, debug } = render(<StockList products={products} setProducts={setProducts} />);
//     debug("Stocklist component");

    const tulip = await getByText('Tulip', { exact: false });
    const daisy = await getByText('Daisy', { exact: false });
    const peony = await getByText('Peony', { exact: false });

    expect(tulip).toBeDefined();
    expect(daisy).toBeDefined();
    expect(peony).toBeDefined();
});

test("Testing that the component does not behave unexpectedly, creating snapshop", () => {
    const tree = render(<StockList products={products} setProducts={setProducts} />).toJSON();
    expect(tree).toMatchSnapshot();
})