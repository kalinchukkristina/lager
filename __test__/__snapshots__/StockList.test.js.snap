// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing that the component does not behave unexpectedly, creating snapshop 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#bbe0f0",
        "fontSize": 20,
        "marginTop": "3%",
      }
    }
  >
    Tulip
     - 
    60
  </Text>
  <Text
    style={
      Object {
        "color": "#bbe0f0",
        "fontSize": 20,
        "marginTop": "3%",
      }
    }
  >
    Daisy
     - 
    202
  </Text>
  <Text
    style={
      Object {
        "color": "#bbe0f0",
        "fontSize": 20,
        "marginTop": "3%",
      }
    }
  >
    Peony
     - 
    35
  </Text>
</View>
`;
